lib/active_model_serializers/key_transform.rb

```diff
@@ -0,0 +1,43 @@
+module ActiveModelSerializers
+  module KeyTransform
+    # Transforms keys to UpperCamelCase or PascalCase.
+    #
+    # @example:
+    #    "some_key" => "SomeKey",
+    # @see: active_support/inflector#camelize
+    def camel(hash)
+      hash.deep_transform_keys! { |key| key.to_s.camelize.to_sym }
+    end
+
+    # Transforms keys to camelCase.
+    #
+    # @example:
+    #    "some_key" => "someKey",
+    # @see: active_support/inflector#camelize
+    def camel_lower(hash)
+      hash.deep_transform_keys! { |key| key.to_s.camelize(:lower).to_sym }
+    end
+
+    # Transforms keys to dashed-case.
+    # This is the default case for the JsonApi adapter.
+    #
+    # @example:
+    #    "some_key" => "some-key",
+    # @see: active_support/inflector#dasherize
+    def dashed(hash)
+      hash.deep_transform_keys! { |key| key.to_s.dasherize.to_sym }
+    end
+
+    # Returns the hash unaltered
+    def unaltered(hash)
+      hash
+    end
+
+    module_function :camel, :camel_lower, :dashed, :unaltered
+
+    def self.method_missing(m, *args, &block)
+      warn "\"#{m}\" is not a defined key transform. Defaulting to :unaltered."
+      send(:unaltered, *args, &block)
```

Comment:
It just seemed nicer to just give back what was supplied and warn if they called an undefined method.